### PR TITLE
Update User-Agent guideline

### PR DESCRIPTION
**Pull request by joheredi**
Updating the General Guidelines with the outcome of the Architecture Board discussion on User-Agent string consistency across languages.

Arch Board issue: https://github.com/Azure/azure-sdk/issues/5110

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

File: docs/general/azurecore.md [hunk 1]
```diff
@@ -33,24 +33,31 @@ In general, the client library will only need to configure these policies.  Howe
 
 Client library usage telemetry is used by service teams (not consumers) to monitor what SDK language, client library version, and language/platform info a client is using to call into their service. Clients can prepend additional information indicating the name and version of the client application.
 
-{% include requirement/MUST id="azurecore-http-telemetry-useragent" %} send telemetry information in the [User-Agent header] using the following format:
+{% include requirement/MUST id="azurecore-http-telemetry-useragent" %} send telemetry information in the [User-Agent header]
```
Comment (JonathanGiles) [inline docs/general/azurecore.md:36]:
```suggestion
{% include requirement/MUST id="azurecore-http-telemetry-useragent" %} send telemetry information in the [User-Agent header].
```

File: docs/general/azurecore.md [hunk 2]
```diff
@@ -33,24 +33,31 @@ In general, the client library will only need to configure these policies.  Howe
...
 
-- (.NET) `AzCopy/10.0.4-Preview azsdk-net-Storage.Blobs/11.0.0 (.NET Standard 2.0; Windows_NT 10.0.14393)`
-- (JavaScript) `AzCopy/10.0.4-Preview azsdk-js-storage-blob/11.0.0 (Node 10.16.0; Ubuntu; Linux x86_64; rv:34.0)`
-- (Java) `AzCopy/10.0.4-Preview azsdk-java-storage.blobs/11.0.0 (Java/1.8.0_45; Macintosh; Intel Mac OS X 10_10; rv:33.0)`
-- (Python) `AzCopy/10.0.4-Preview azsdk-python-storage/4.0.0 Python/3.7.3 (Ubuntu; Linux x86_64; rv:34.0)`
+- (.NET) `AzCopy/10.0.4-Preview azsdk-net-Storage.DataLake/12.0.0  azsdk-net-Storage.Blobs/12.0.0 azsdk-net-Core/11.0.0 (.NET Standard 2.0; Windows_NT 10.0.14393)`
+- (JavaScript) `AzCopy/10.0.4-Preview azsdk-js-storage-datalake/12.0.0 azsdk-js-storage-blob/12.0.0 azsdk-js-core-client/1.0.0 azsdk-js-core-rest-pipeline/11.0.0 (Node 18.12.1; Ubuntu; Linux x86_64; rv:34.0)`
+- (Java) `AzCopy/10.0.4-Preview azsdk-java-storage.datalake/12.0.0 azsdk-java-storage.blobs/12.0.0 azsdk-java-core/11.0.0 (Java/1.8.0_45; Macintosh; Intel Mac OS X 10_10; rv:33.0)`
```
Comment (JonathanGiles) [inline docs/general/azurecore.md:59]:
```suggestion
- (Java) `AzCopy/10.0.4-Preview azsdk-java-storage-datalake/12.0.0 azsdk-java-storage-blobs/12.0.0 azsdk-java-core/11.0.0 (Java/1.8.0_45; Macintosh; Intel Mac OS X 10_10; rv:33.0)`
```

**Review by JonathanGiles — CHANGES_REQUESTED** [review]
One nit, one Java-specific typo that should be fixed.

File: docs/general/azurecore.md [hunk 3]
```diff
@@ -33,24 +33,31 @@ In general, the client library will only need to configure these policies.  Howe
 
 Client library usage telemetry is used by service teams (not consumers) to monitor what SDK language, client library version, and language/platform info a client is using to call into their service. Clients can prepend additional information indicating the name and version of the client application.
 
-{% include requirement/MUST id="azurecore-http-telemetry-useragent" %} send telemetry information in the [User-Agent header] using the following format:
+{% include requirement/MUST id="azurecore-http-telemetry-useragent" %} send telemetry information in the [User-Agent header].
 
+{% include requirement/MUST id="azurecore-http-telemetry-useragent-stacking" %} extend the user agent to list all `client wrapping` dependencies, sorting them from outermost to innermost.
```
Comment (KrzysztofCwalina) [inline docs/general/azurecore.md:38]: I would add an example here. Without it, I suspect many readers will not know what 'client wrapping' means

File: docs/general/azurecore.md [hunk 4]
```diff
@@ -33,24 +33,31 @@ In general, the client library will only need to configure these policies.  Howe
...
+
+{% include requirement/SHOULD id="azurecore-http-telemetry-useragent-stacking-core" %} include core dependencies in the user agent.
```
Comment (KrzysztofCwalina) [inline docs/general/azurecore.md:40]: what are "core dependencies"? i.e. we need to be more precises such that a reader can actually implement this guideline without having to ask us how.

**Pull request merged**